src/plugins/query_enhancements/public/search/sql_search_interceptor.ts

```diff
@@ -46,7 +46,11 @@ export class SQLSearchInterceptor extends SearchInterceptor {
       },
     };
 
-    return fetch(context, this.queryService.queryString.getQuery()).pipe(
+    // Use query from request if available, otherwise fall back to queryStringManager
+    const query =
+      request.params?.body?.query?.queries?.[0] || this.queryService.queryString.getQuery();
```

Comment:
tbh. i dont think there should be a case where the search interceptor shouldnt get the query object.